Mathlib/Algebra/Ring/Idempotent.lean

```diff
@@ -76,3 +77,39 @@ lemma add_sub_mul (hp : IsIdempotentElem a) (hq : IsIdempotentElem b) :
 
 end CommRing
 end IsIdempotentElem
+
+lemma isIdempotentElem_sub_of {H : Type*} [NonAssocRing H] {p q : H}
```

Comment:
Why not reuse the variable `R` from the above? And you also might consider moving these results into the `IsIdempotentElem` namespace above to allow for dot-notation.